### PR TITLE
[58] Develop an E-commerce Store Database

### PR DESCRIPTION
**Pull request by ainobs**
## Task: `Name of given task`
- `Changes you made to given task`

### CODE REVIEW

**Review by adaugochi — CHANGES_REQUESTED** [review]
Well done @ainobs 

Kindly rearrange your queries. E.g, since category_id is used as a foreign key on the products table it, therefore, means that the categories table need to be created first before products table.

Also, tables should have an `id` field. Don't used this format `tablename_id` to represent the `id` of a table

**Comment by ainobs** [comment]
> Well done @ainobs
> 
> Kindly rearrange your queries. E.g, since category_id is used as a foreign key on the products table it, therefore, means that the categories table need to be created first before products table.
> 
> Also, tables should have an `id` field. Don't used this format `tablename_id` to represent the `id` of a table

Thanks for the review. The changes have been effected

File: submissions/victorAienobe/database/init.sql [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	PRIMARY KEY (id)
+);
+
+-- created the products table
+CREATE TABLE products (
+	id INT NOT NULL AUTO_INCREMENT,
+	name VARCHAR(50) NOT NULL,
+	description VARCHAR(250) NOT NULL,
+	image BLOB,
+    category_id INT,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:28]: should be `NOT NULL`

File: submissions/victorAienobe/database/init.sql [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    admin_id INT,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:29]: same here

File: submissions/victorAienobe/database/init.sql [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	country VARCHAR(20) NOT NULL,
+	phone_number BIGINT NOT NULL,
+    PRIMARY KEY (id),
+	FOREIGN KEY (customer_id) REFERENCES customers (id) ON DELETE CASCADE
+);
+
+-- Created the orders table
+CREATE TABLE orders (
+	id INT NOT NULL AUTO_INCREMENT,
+	customer_orders INT,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:65]: I think `customer_id` is fine here

File: submissions/victorAienobe/database/init.sql [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	order_amount DECIMAL(7,2) NOT NULL,
+	order_created_at DATETIME,
+	PRIMARY KEY (id),
+	FOREIGN KEY (customer_orders) REFERENCES customers (id) ON DELETE CASCADE
+);
+-- Auto increment count is set to start from 100
+ALTER TABLE orders AUTO_INCREMENT = 100;
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:72]: Is there a reason for this?

File: submissions/victorAienobe/database/init.sql [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+-- Created the order items table
+CREATE TABLE order_items (
+	id INT NOT NULL AUTO_INCREMENT,
+	order_id INT,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:77]: should be `NOT NULL`. or is there a reason why you want it to be `NULL`. cos by default, that is what it is

File: submissions/victorAienobe/database/init.sql [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	product_id INT,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:78]: same here

File: submissions/victorAienobe/database/init.sql [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+-- Inserted values into the admins table
+INSERT INTO admins (name, email, password, phone_number) 
+VALUES ('victor aienobe', 'viktornobe@gmail.com', 'smoke&mirrors', +2348108088144),
+('benedit ejike', 'ben@gmail.com', 'thanos', +2349025725272),
+('chidi okpala', 'chidi@gmail.com', 'drstrange', +2347026372527),
+('simon alvin peters', 'alvinpeters@gmail.com', 'aryastark', +2348025745262);
+
+-- Inserted values into the products table
+INSERT INTO products (name, description, image, category_id, admin_id, unit_price, stock_level, status)
+VALUES ('blue white collar tee shirt', 'This is a quality shirt you can put on all day', LOAD_FILE('C:\Users\Aienobe\Desktop\Dufuna-CodeCamp20\submissions\victorAienobe\html\images\shirts.png'), 2, 1, '250', '70', 'in stock'),
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:103]: I will advise you use online image URL

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/victorAienobe/database/init.sql [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+-- created the products table
+CREATE TABLE products (
+	id INT NOT NULL AUTO_INCREMENT,
+	name VARCHAR(50) NOT NULL,
+	description VARCHAR(250) NOT NULL,
+	image BLOB,
+    category_id INT NOT NULL,
+    admin_id INT NOT NULL,
+	unit_price INT NOT NULL,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:30]: `price`  or `amount` should be `DECIMAL` and not `INT`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/victorAienobe/database/init.sql [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+-- created the products table
+CREATE TABLE products (
+	id INT NOT NULL AUTO_INCREMENT,
+	name VARCHAR(50) NOT NULL,
+	description VARCHAR(250) NOT NULL,
+	image BLOB,
+    category_id INT NOT NULL,
+    admin_id INT NOT NULL,
+	unit_price DECIMAL(4,0) NOT NULL,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:30]: Is this not too small?

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by ainobs** [comment]
It covers the range of values I inserted into the table...

File: submissions/victorAienobe/database/init.sql [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+-- Created the customer addresses table
+CREATE TABLE customer_addresses (
+	id INT NOT NULL AUTO_INCREMENT,
+	customer_id INT NULL,
+	street_address VARCHAR(100) NOT NULL,
+	city VARCHAR(20) NOT NULL,
+	state VARCHAR(20) NOT NULL,
+	zip_code INT,
+	country VARCHAR(20) NOT NULL,
+	phone_number BIGINT NOT NULL,
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:57]: phone_number should be a VARCHAR

File: submissions/victorAienobe/database/init.sql [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+-- Inserted values into the categories table
+INSERT INTO categories (category_name)
+VALUES ('game console'),
+('Mens-shirt'),
+('soccer boots'),
+('club jerseys');
+
+-- Inserted values into the admins table
+INSERT INTO admins (name, email, password, phone_number) 
+VALUES ('victor aienobe', 'viktornobe@gmail.com', 'smoke&mirrors', +2348108088144),
```
Comment (adaugochi) [inline submissions/victorAienobe/database/init.sql:94]: phone_number should be wrapped in quotes. The `+` sign in front of it should make it be a string and not  INT

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/victorAienobe/database/init.sql (hunk 11 above)
Comment (ainobs) [reply submissions/victorAienobe/database/init.sql]: thanks.. corrections have been made

**Review by adaugochi — APPROVED** [review]

**Pull request merged**